### ISSUE TITLE
Required property "spec" is not present in dictionary metadata

### ISSUE THREAD

**Issue by adnejacobsen** [equinor/fmu-dataio#365]
Json schema validation in Sumo fails when attempting to upload metadata for parameters dictionary, required property "spec" is not present.

**Comment by adnejacobsen** [comment]
> What should `spec` be for a dictionary...?

Looks like `spec` and `bbox` are set to empty dictionaries in the code, but the resulting metadata is missing both properties.

**Comment by perolavsvendsen** [comment]
Yes, empties are removed. I don't think `bbox` is required property. But spec might be. It should probably be required for e.g. `table` and things like that, but this should perhaps be a conditional requirement, not a general one.

**Comment by adnejacobsen** [comment]
Yes, `spec` is required and `bbox` is not:

```json
  "data": {
      "type": "object",
      "title": "The data block",
      "required": [
          "content",
          "name",
          "format",
          "spec",
          "stratigraphic",
          "is_prediction",
          "is_observation"
      ]
```

I will look into making some changes to the schema in #361. For which object types should we require `spec`?

```json
  "surface",
  "table",
  "cpgrid",
  "cpgrid_property",
  "polygons",
  "cube",
  "well",
  "points",
```

**Comment by perolavsvendsen** [comment]
At least for `table`, but possibly also for `surface`. @jcrivenaes opinions?

**Comment by perolavsvendsen** [comment]
...or are there any logical `specs` for a dict structure? I cannot think of any, at least.

**Comment by adnejacobsen** [comment]
The only thing that comes to mind is `spec.keys`. Not sure if if provides any value and we could end up with a really long lists for large dictionaries.
